recipe/meta.yaml

```
@@ -1,78 +1,72 @@
-    - CUDA_HOME  # [ppc64le or aarch64]
+  # this recipe only supports CUDA 12; the other CUDA versions are in the cuda-11 branch
+  skip: true  # [win32 or osx]
   script:
     - mkdir -p $PREFIX/include                                             # [linux]
     - mv include/* $PREFIX/include/                                        # [linux]
     - mkdir -p $PREFIX/lib                                                 # [linux]
-    - mv lib/{{ cuda_compiler_version }}/*.so* $PREFIX/lib/                # [linux and cuda_compiler_version in ("10.2", "11.0")]
-    - mv lib/11/*.so* $PREFIX/lib/                                         # [linux and cuda_compiler_version == "11.1"]
+    - mv lib/{{ cuda_major }}/*.so* $PREFIX/lib/                           # [linux]
     - patchelf --add-needed libcudart.so $PREFIX/lib/libcutensor.so        # [ppc64le]
 
     - copy include\\cutensor.h %LIBRARY_INC%\\                             # [win64]
     - copy include\\cutensorMg.h %LIBRARY_INC%\\                           # [win64]
     - mkdir %LIBRARY_INC%\\cutensor                                        # [win64]
     - copy include\\cutensor\\types.h %LIBRARY_INC%\\cutensor              # [win64]
-    - del lib\\{{ cuda_compiler_version }}\\*static*                       # [win64 and cuda_compiler_version in ("10.2", "11.0")]
-    - copy lib\\{{ cuda_compiler_version }}\\*.dll %LIBRARY_BIN%\\         # [win64 and cuda_compiler_version in ("10.2", "11.0")]
-    - copy lib\\{{ cuda_compiler_version }}\\*.lib %LIBRARY_LIB%\\         # [win64 and cuda_compiler_version in ("10.2", "11.0")]
-    - del lib\\11\\*static*                                                # [win64 and cuda_compiler_version in ("11.1", )]
-    - copy lib\\11\\*.dll %LIBRARY_BIN%\\                                  # [win64 and cuda_compiler_version in ("11.1", )]
-    - copy lib\\11\\*.lib %LIBRARY_LIB%\\                                  # [win64 and cuda_compiler_version in ("11.1", )]
-  ignore_run_exports:
-    - cudatoolkit
+    - del lib\\{{ cuda_major }}\\*static*                                  # [win64]
+    - copy lib\\{{ cuda_major }}\\*.dll %LIBRARY_BIN%\\                    # [win64]
+    - copy lib\\{{ cuda_major }}\\*.lib %LIBRARY_LIB%\\                    # [win64]
   run_exports:
     - {{ pin_subpackage('cutensor') }}
   missing_dso_whitelist:
-    # suppress warning, as these are included in the run dependency
-    - '*/libcublasLt.so*'  # [linux and cuda_compiler_version in ("11.0", "11.1")]
-    - '*/cublasLt64*.dll'  # [win64 and cuda_compiler_version in ("11.0", "11.1")]
+    - "*/libcuda.so.*"    # [linux]
+    - "*/nvcuda.dll"      # [win]
 
 requirements:
   build:
     - {{ compiler('c') }}
-    - {{ compiler('cuda') }}
-    - sysroot_linux-64 2.17  # [linux64]
+    - {{ compiler('cxx') }}
+    - arm-variant * {{ arm_variant_type }}  # [aarch64]
+    - sysroot_{{ target_platform }} 2.17    # [linux64]
```

I think `sysroot` should be needed on ARM too (`linux64` is just `x86_64`). The CUDA 12 recipes are using `linux` here.
```suggestion
    - sysroot_{{ target_platform }} 2.17    # [linux]
```

---

Good catch!